rust_src/src/fonts.rs

```diff
@@ -97,6 +103,70 @@ impl FontExtraType {
     }
 }
 
+pub type LispFontObjectRef = ExternalPtr<Lisp_Font_Object>;
+
+impl LispFontObjectRef {
+    pub fn as_lisp_obj(self) -> LispObject {
+        LispObject::tag_ptr(self, Lisp_Type::Lisp_Vectorlike)
+    }
+
+    pub fn add_log(self, action: &str, result: LispObject) {
+        let c_str = CString::new(action).unwrap();
+        unsafe { font_add_log(c_str.as_ptr(), self.as_lisp_obj(), result) }
+    }
+
+    pub fn close(mut self, mut frame: LispFrameRef) {
+        if data::aref(self.as_lisp_obj(), FONT_TYPE_INDEX.into()).is_nil() {
+            // Already closed
```

Comment:
Instead of the aref another option is to convert to vectorlike and use get. I am not sire which is best.

Reply:
I just copied the behavior form the c function `font_close_object`, I'll see if the other way is nicer